File: Day 1/main.py
def part_one():
    counter = 0
    with open("input.txt") as file:
        lines = file.readlines()
        measurements = [int(line) for line in lines]
        for i in range(1, len(measurements)):
            if (measurements[i] > measurements[i - 1]):
                counter += 1
    print(f"{counter}")

def part_two():
    counter = 0
    with open("input.txt") as file:
        lines = file.readlines()
        measurements = [int(line) for line in lines]
        for i in range(0, len(measurements) - 3):
            first_window = measurements[i] + measurements[i + 1] + measurements[i + 2]
            second_window = measurements[i + 1] + measurements[i + 2] + measurements[i + 3]
            if (second_window > first_window):
                counter += 1
    print(f"{counter}")

if __name__ == "__main__":
    part_one()
    part_two()

File: Day 1/rust/src/main.rs
fn read_lines() -> Vec<i32> {
    let file = include_str!("../../input.txt");
    file.lines()
        .map(|line| line.parse::<i32>().unwrap())
        .collect()
}

fn part_one() {
    let mut counter = 0;
    let measurements = read_lines();
    for i in 1..measurements.len() {
        if measurements[i] > measurements[i - 1] {
            counter += 1;
        }
    }
    println!("{}", counter);
}

fn part_two() {
    let mut counter = 0;
    let measurements = read_lines();
    for i in 0..(measurements.len() - 3) {
        let first_window = measurements[i] + measurements[i + 1] + measurements[i + 2];
        let second_window = measurements[i + 1] + measurements[i + 2] + measurements[i + 3];
        if second_window > first_window {
            counter += 1;
        }
    }
    println!("{}", counter);
}

fn main() {
    part_one();
    part_two();
}
